### ISSUE TITLE
[Bug] Text size isn't set up properly when Android display size is set to the largest

### ISSUE THREAD

**Issue by kelimuttu** [mozilla-mobile/fenix#18226]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
- Set Android text display size to the largest from your phone Setting
- Open Firefox
- Open the tab tray (list or grid view)

### Expected behavior
- Tabs should be displayed properly. 

### Actual behavior
- Website title's on the tab tray aren't properly displayed. 

### Device information
MIUI 12.0.3 - Android 10

* Device vendor / model and Android version: (Reproducible on Huawei P30 Pro - EMUI version 10.1.0 - Android 10 )
* Firefox for Android version: Firefox 86.1.1 (Build #2015794883)

**Comment by lobontiumira** [comment]
I can confirm this issue on Samsung Galaxy Note 8 (Android 9), on all Firefox builds, RC 87.0.0, Beta 88.0.0-beta.2, and 3/31 Nightly.

@kelimuttu thank you for your report!

**Comment by neha-b2001** [comment]
@lobontiumira Can I please work on this issue?

**Comment by lobontiumira** [comment]
It is not my decision.

**Comment by neha-b2001** [comment]
Thanks @eliserichards 👍

**Comment by testuser51** [comment]
there is such a problem here too. Same steps to reproduce
![IMG_20210421_095806](https://user-images.githubusercontent.com/53906414/115510418-60e7fb80-a288-11eb-8609-aabbfc711a72.jpg)
![IMG_20210421_095826](https://user-images.githubusercontent.com/53906414/115510423-62192880-a288-11eb-97f3-5921c50f4644.jpg)

**Comment by jonalmeida** [comment]
@testuser51 I've filed #19271 as a separate issue for this.

**Comment by lobontiumira** [comment]
Verified as fixed on the 4/28 Nightly build with Sony Xperia Z5 Premium (Android 7.1.1), Google Pixel (Android 10), and HTC 10 (Android 8).

<img src="https://user-images.githubusercontent.com/40694158/116408648-bd748900-a83b-11eb-89a4-12054558bcb6.png" width="300" height="550" />

**Comment by eliserichards** [comment]
Let's do another round of QA on this now that the PR is merged. Can we test on on more than one device?

**Comment by SoftVision-LorandJanos** [comment]
This issue is no longer reproducible on the latest Nightly (90.0a1 - 5/10 @ 5:07AM), and on the latest Beta (89.0.0-beta.5). On the latest Release (88.1.3) it is still reproducible.

Devices used: HTC Nexus 9 with Android 7.1.1 and Motorola Nexus 6 with Android 7.1.1.

Example:
![Screenshot_20210510-171833](https://user-images.githubusercontent.com/48829350/117675742-82684300-b1b5-11eb-9eb8-89081c99528c.png)
![Screenshot_20210510-171951](https://user-images.githubusercontent.com/48829350/117675737-81371600-b1b5-11eb-9960-38cd1d7927cd.png)
